ludwig/utils/triton_utils.py

```diff
@@ -49,142 +93,522 @@ def forward(self, {input_signature}):
+        )
+        return config
+
+
+@dataclass
+class TritonModel:
+    """Enables the scripting and export of a model.
+
+    :param module: the inference module.
+    :param input_features: input features of the model.
+    :param output_features: output features of the model.
+    :param inference_stage: one of PREPROCESSOR, PREDICTOR, POSTPROCESSOR.
+    """
+
+    module: Union[_InferencePreprocessor, _InferencePredictor, _InferencePostprocessor]
+    input_features: List[TritonConfigFeature]
+    output_features: List[TritonConfigFeature]
+    inference_stage: str
+
+    def _get_dict_type_hint(self) -> str:
+        return {
+            PREPROCESSOR: "TorchscriptPreprocessingInput",
+            PREDICTOR: "torch.Tensor",
+            POSTPROCESSOR: "torch.Tensor",
+        }[self.inference_stage]
+
+    def _get_input_signature(self, triton_features: List[TritonConfigFeature]) -> str:
+        elems = [
+            f"{feature.wrapper_signature_name}: {feature._get_wrapper_signature_type()}" for feature in triton_features
+        ]
+        return ", ".join(elems)
+
+    def _get_input_dict(self, triton_features: List[TritonConfigFeature]) -> str:
+        elems = [f'"{feature.name}": {feature.wrapper_signature_name}' for feature in triton_features]
+        return "{" + ", ".join(elems) + "}"
+
+    def _get_output_tuple(self, triton_features: List[TritonConfigFeature]) -> str:
+        elems = [f'results["{feature.name}"]' for feature in triton_features]
+        return "(" + ", ".join(elems) + ",)"
+
+    def generate_inference_module_wrapper(self) -> str:
+        """Generate the class wrapper around an inference module."""
+        return INFERENCE_MODULE_TEMPLATE.format(
+            input_signature=self._get_input_signature(self.input_features),
+            input_type=self._get_dict_type_hint(),
+            input_dict=self._get_input_dict(self.input_features),
+            output_tuple=self._get_output_tuple(self.output_features),
+        )
+
+    def generate_scripted_module(self):
+        """Generate the scripted module from the wrapper class."""
+        wrapper_definition = self.generate_inference_module_wrapper()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            ts_path = os.path.join(tmpdir, "generated.py")
+            with open(ts_path, "w") as f:
+                f.write(wrapper_definition)
+
+            spec = importlib.util.spec_from_file_location("generated.ts", ts_path)
+            gen_ts = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(gen_ts)
+
+            gen_module = gen_ts.GeneratedInferenceModule(self.module)
+            scripted_module = torch.jit.script(gen_module)
+        return scripted_module
 
 
 def export_triton(
-    model: LudwigModel, output_path: str, model_name: str = "ludwig_model", model_version: Union[int, str] = 1
-) -> Tuple[str, str]:
+    model: LudwigModel,
+    data_example: pd.DataFrame,
```

Comment:
If possible would be nice to make `data_example` optional, or not required given we can generate a dataset from the config eg:

```
        from ludwig.data.dataset_synthesizer import build_synthetic_dataset
        import pandas as pd
        import shutil
        import tarfile
        def generate_df(config, num_examples=2):
            features = config['input_features'] + config['output_features']
            df = build_synthetic_dataset(num_examples, features)
            data = [next(df) for _ in range(num_examples)]
            return pd.DataFrame(data[1:], columns=data[0])
      try:
          example_df = generate_df(model.config)
          output_paths: Dict[str, Tuple[str, str]] = export_triton(model, example_df, tmpdir,
              model_name, model_version, device, device_count)
          logger.info("Got paths", output_paths)
      except Exception as e:
          msg = f"error in triton export for model {model_name}"
          logger.error(msg, e)
          raise Exception(msg)
      finally:
          # Clean up any temporay files that might have been created
          shutil.rmtree('audio_files', ignore_errors=True)
          shutil.rmtree('image_files', ignore_errors=True)

```